test/CMakeLists.txt

```diff
@@ -47,6 +47,13 @@ if (BUILD_TESTS)
                 WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/cfg)
     endif()
 
+    # TODO: the target name "test" is reservered by CTest
+    #add_custom_target(test $<TARGET_FILE:testrunner>
+    #        DEPENDS cppcheck testrunner)
+
+    add_custom_target(check $<TARGET_FILE:testrunner> -q
```

Comment:
I dont understand the purpose of this, as ctest will already run the tests in the testrunner and it will run faster. Can we name this `run-testrunner` instead?

Reply:
First I want to align `Makefile` with CMake so there's an easy transition when moving to CMake. The `Makefile` should be completely deprecated or stripped down a bit.

There's lots of more things in CMake which can be simplified and consolidated.

Reply:
> First I want to align Makefile with CMake so there's an easy transition when moving to CMake.

I think only @danmar uses the Makefile(and he seems more focused on cppcheck premium recently too). As far as I know, most other developers and contributors use cmake. 

We can definitely provide targets to do the same functionality as in the makefile, but I dont see the need to name them exactly the same, especially when the name already overlaps with targets developers are already using in cmake.